packages/rocketchat-language-plugin/server/server.js

```diff
@@ -0,0 +1,82 @@
+const parser = Npm.require('accept-language-parser');
+const languages = Npm.require('languages');
+
+function get_language(languages_codes) {
+
+	const languages_parsed = parser.parse(languages_codes);
+	if (languages_parsed.length===0) {
+		return;
+	}
+	const priority_language = languages_parsed[0].code;
+
+	const priority_language_fullname = languages.getLanguageInfo(priority_language);
+	return priority_language_fullname;
+}
+
+function remove_user_from_automatic_channel() {
+	const user_info = RocketChat.models.Users.findOneById(Meteor.userId());
+	// get the name of channels to which user is added
+	const to_be_removed = user_info.automatic_channels.filter(function(obj) {
+		return obj.blacklisted ==='false';
+	});
+	if (to_be_removed.length!==0) {
```

Comment:
I am not finding findAndModify much useful in this way, because it returns the whole document and then we again need to search the array ``` automatic_channels```. What if we go this way:
```
function remove_user_from_automatic_channel(user, channel) {
	
		const document = Meteor.users.update({
			_id: user._id
		}, {
			$pull: { automatic_channels: { blacklisted: false } }
		});
		if (document) {
			user.automatic_channels.forEach( function (arrayItem) {
			if(arrayItem.name!==channel && arrayItem.blacklisted===false) {
			const room = RocketChat.models.Rooms.findOneByIdOrName(arrayItem.name);
			RocketChat.removeUserFromRoom(room._id, user);
```